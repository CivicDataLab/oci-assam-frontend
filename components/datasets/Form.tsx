import { useState } from 'react';
import { useRouter } from 'next/router';

const Form: React.FC = () => {
  const router = useRouter();
  const [q, setQ] = useState(router.query.q);
  const [sort, setSort] = useState(router.query.sort);

  const handleChange = (event) => {
    switch (event.target.name) {
      case 'q':
        setQ(event.target.value);
        break;
      case 'sort':
        setSort(event.target.value);
        router.push({
          pathname: '/datasets',
          query: { q, sort: event.target.value },
        });
        break;
    }
  };

  function handleClear() {
    const input = document.querySelector('.form__input') as HTMLInputElement;
    input.value = '';
    input.focus();
  }

  const handleSubmit = (event) => {
    event.preventDefault();
    router.push({
      pathname: '/datasets',
      query: { q, sort },
    });
  };

  return (
    <form onSubmit={handleSubmit} className="form">
      <div className="form__search">
        <input
          type="search"
          name="q"
          value={q}
          onChange={handleChange}
          placeholder="Search here"
          aria-label="Search"
          className="form__input"
        />
        <div className="form__buttons">
          <button
            className="form__clear"
            type="button"
            title="Clear search field"
            onClick={handleClear}
          >
            <span className="sr-only">Clear search field</span>
            &#x2715;
          </button>
          <button
            onClick={handleSubmit}
            className="form__submit"
            type="submit"
            title="Submit search"
          >
            <span className="sr-only">Submit search</span>
            &#x279C;
          </button>
        </div>
      </div>
      <div className="form__sort">
        <label htmlFor="field-order-by" className="sr-only">
          Sort by:&nbsp;
        </label>
        <select
          className="form__select select-comp"
          id="field-order-by"
          name="sort"
          onChange={handleChange}
          onBlur={handleChange}
          value={sort}
        >
          <option value="score:desc">Relevance</option>
          <option value="title_string:asc">Name Ascending</option>
          <option value="title_string:desc">Name Descending</option>
          <option value="metadata_modified:desc">Last Modified</option>
          <option value="views_recent:desc">Popular</option>
        </select>
      </div>
    </form>
  );
};

export default Form;
