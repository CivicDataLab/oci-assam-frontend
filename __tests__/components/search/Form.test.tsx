import React from 'react';
import { render } from '@testing-library/react';
import Form from '../../../components/search/Form';

const useRouter = jest.spyOn(require('next/router'), 'useRouter');

test('ðŸ“¸ of Form component with empty', () => {
  useRouter.mockImplementationOnce(() => ({
    query: { search: '', sort: '', size: '' },
  }));

  const { container } = render(<Form />);
  expect(container).toMatchSnapshot();
});

// test('ðŸ“¸ of Form component with query', () => {
//   useRouter.mockImplementationOnce(() => ({
//     query: { search: 'gdp', sort: '' },
//   }));

//   const { container } = render(<Form />);
//   expect(container).toMatchSnapshot();
// });
