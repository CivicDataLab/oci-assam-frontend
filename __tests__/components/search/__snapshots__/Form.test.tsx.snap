// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ“¸ of Form component with empty 1`] = `
<div>
  <form
    class="items-center"
  >
    <div
      class="flex"
    >
      <input
        aria-label="Search"
        class="bg-white focus:outline-none focus:shadow-outline border border-gray-300 w-1/2 rounded-lg py-2 px-4 block appearance-none leading-normal"
        name="q"
        placeholder="Search"
        type="text"
        value=""
      />
      <button
        class="inline-block text-sm px-4 py-3 mx-3 leading-none border rounded text-white bg-black border-black lg:mt-0"
      >
        Search
      </button>
    </div>
    <div
      class="inline-block my-6"
    >
      <label
        for="field-order-by"
      >
        Items per page:
      </label>
      <select
        class="bg-white"
        id="field-order-by"
        name="size"
      >
        <option
          value="10"
        >
          10
        </option>
        <option
          value="20"
        >
          20
        </option>
        <option
          value="50"
        >
          50
        </option>
      </select>
    </div>
    <div
      class="inline-block my-6 float-right"
    >
      <label
        for="field-order-by"
      >
        Order by:
      </label>
      <select
        class="bg-white"
        id="field-order-by"
        name="sort"
      >
        <option
          value="score:desc"
        >
          Relevance
        </option>
        <option
          value="title_string:asc"
        >
          Name Ascending
        </option>
        <option
          value="title_string:desc"
        >
          Name Descending
        </option>
        <option
          value="metadata_modified:desc"
        >
          Last Modified
        </option>
        <option
          value="views_recent:desc"
        >
          Popular
        </option>
      </select>
    </div>
  </form>
</div>
`;
